cpp/include/kvikio/defaults.hpp

```diff
@@ -109,6 +110,14 @@ class defaults {
       }
       _task_size = env;
     }
+    // Determine the default value of `gds_threshold`
+    {
+      const ssize_t env = detail::getenv_or("KVIKIO_GDS_THRESHOLD", 1024 * 1024);
```

Comment:
If this an empirically determined "good" default? How does it align with the task size setting?

Reply:
Setting `KVIKIO_GDS_THRESHOLD < KVIKIO_TASK_SIZE` covers the case where the buffer is large enough to take advantage of cuFile and GDS but not large enough to benefit from concurrent IO. 
Having said that, I haven't done extensive benchmarking so better defaults probably exist :)